templates/rbac-clusterrole.yaml

```diff
@@ -0,0 +1,188 @@
+# We overwrite system:node, rather than patching
+# due to a race condition where certain services
+# are expecting RBAC to be ready.
```

Comment:
Is this necessary? I would expect to be able to create a new ClusterRole (possibly with a different name) that just grants "get namespace", and bind it to the system:nodes group, without clobbering existing permissions.

I'll look into this. Let me brush up on my understanding of RBAC and I'll get back to you.

Reply:
That wasn't working for me when trying; it'd overwrite `system:node`.  It's more than likely I was doing something wrong though :)

Reply:
This worked for me: https://gist.github.com/Cynerva/5b5389581069c9e8ae9641acc490cb24

The different ClusterRole name (`namespace-reader` instead of `system:node`) is key there. This allows us to grant the "get namespace" permission via a secondary ClusterRole, without redefining the existing `system:node` one.

I think it's important that we do this. Overriding the `system:nodes` ClusterRole with our own definition may give us trouble down the line, if upstream k8s changes the default set of permissions for `system:node` but our charm overrides it with an older set.

Reply:
Ah cool, I thought that it was required to be named `system:node` as well.  That's a much nicer fix.